src/encoded/static/components/item-pages/components/TabbedView.js

```diff
@@ -68,7 +68,12 @@ export class TabbedView extends React.PureComponent {
     }
 
     static renderTabPane(tabObj, tabIndex = 0){
-        const { key, tab, placeholder, disabled = false, style = null, content } = tabObj;
+        const { key, tab, placeholder, disabled = false, style = null } = tabObj;
+        let { content } = tabObj;
+        if(this.hiGlassTabIndex >= 0){
```

Comment:
1. `renderTabPane` is a static method where `this` refers to the _function itself_ and not any class instance. `this.hiGlassTabIndex` will always evaluate to `undefined` and condition would always evaluate to false.

2. If it _were_ an instance method, would try to use `memoize` library instead for ease of reasoning. Assigning data to `this` (especially without using `setState`) can get very complex/unwieldy quickly so is nice to avoid if possible. If were an instance method, could perhaps be refactored along these lines:
    ```
    static getHiGlassTabIndex(contents){
        return _.findIndex(contents, function (el) { return el.key === "higlass" });
    }
    
    constructor(props){
        ...
        this.memoized = {
            // Maybe - perhaps unnecessary to memoize since not too unperformant
            getHiGlassTabIndex : memoize(TabbedView.getHiGlassTabIndex)
        }
        ...
    }

    renderTabPane(){ // IF INSTANCE METHOD
        ...
        const higlassTabIndex = this.memoized.getHiGlassTabIndex(contents)
    }
    ```
However this is moot anyway since I guess not needed here and changes in HiGlassViewConfigView.js should be sufficient (haven't tested yet). Other consideration is tab-specific or ItemView-specific logic should ideally not live in TabbedView.js or other 'global'/'reusable' places.

Reply:
Re: 1. `renderTabPane` ...

@alexkb0009  Yes, 100% agree with you that adding hiGlassTabIndex creates a spaghetti code. On the other hand, _.map function's third parameter is context (https://underscorejs.org/#map) and sets this of TabbedView.renderTabPane function.

`'children': _.map(allTabs, TabbedView.renderTabPane, { 'hiGlassTabIndex': hiGlassTabIndex })`

this.hiGlassTabIndex in renderTabPane actually returns an integer. By the way, this truth  
not helps to make the code more readable. We'll remove it asap.
```
if(this.hiGlassTabIndex >= 0){
            content = React.cloneElement(content, { 'hiGlassTabIndex': this.hiGlassTabIndex });
}
```